src/particles/diagnostics/DiagnosticOutput.cpp

```diff
@@ -53,43 +54,51 @@ namespace impactx::diagnostics
         int const nLevel = tmp.finestLevel();
         for (int lev = 0; lev <= nLevel; ++lev) {
             // loop over all particle boxes
-            using ParIt = typename decltype(tmp)::ParConstIterType;
+            using ParIt = amrex::ParIterSoA<RealSoA::nattribs, IntSoA::nattribs>;
```

Comment:
This does not need to be changed here.

But the line above
```c++
auto tmp = pc.make_alike<amrex::PinnedArenaAllocator>();
```
needs to be implemented in AMReX still: https://github.com/AMReX-Codes/amrex/pull/2878#issuecomment-1249975075

Reply:
Ok let's do this